alicloud/resource_alicloud_ddosbgp_instance.go

```diff
@@ -0,0 +1,203 @@
+package alicloud
+
+import (
+	"strconv"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/bssopenapi"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/ddosbgp"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func resourceAlicloudDdosbgpInstance() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudDdosbgpInstanceCreate,
+		Read:   resourceAlicloudDdosbgpInstanceRead,
+		Update: resourceAlicloudDdosbgpInstanceUpdate,
+		Delete: resourceAlicloudDdosbgpInstanceDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Required:     false,
+				Default:      "1",
+				ForceNew:     true,
+				ValidateFunc: validateDdosbgpInstanceType,
```

Comment:
Please use validateAllowedStringValue to replace `validateDdosbgpInstanceType `.